mvpa2/measures/label.py

```diff
@@ -0,0 +1,139 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the PyMVPA package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Label adjacent features in a dataset
+"""
+import numpy as np
+from collections import defaultdict
+from operator import itemgetter
+
+from ..base.param import Parameter
+from ..base.constraints import EnsureBool
+from ..datasets.base import Dataset
+from ..misc.neighborhood import CachedQueryEngine
+
+from .base import Measure
+
+if __debug__:
+    from ..base import debug
+
+
+class ClustersLabeler(Measure):
+    """Determines contiguous regions of features using a query engine.
+
+    "label" in the name chosen to mimic scipy.ndimage.measurements.label
+    """
+    qe = Parameter(
+        None,
+        doc="QueryEngine to determine neighbors of any given feature")
```

Comment:
Do you mean using the QE used to create the searchlight map?
It seems that there would be cases when multiple clusters are considered a single cluster just due to extended connectivity definition.  See the attached schema with thresholded map in blue and neighborhood in circles overlay.
![slght](https://cloud.githubusercontent.com/assets/1155388/14827096/c1105618-0bae-11e6-9b54-32c413c3e81a.png)
In that case it would certainly overestimate the null cluster size distribution by merging noisy patches in permutated maps, but also overestimate the cluster size in the true map by created superclusters?

Also when using surface QE, the neighborhood might vary for each subject for each location due to different distances between vertices of a standard surface spherically warped to the subject space. Which surface would you then use for group analysis, the template one (fsaveragex/icox)?